src/main/java/edu/harvard/iq/dataverse/DataverseRoleServiceBean.java

```diff
@@ -303,7 +303,7 @@ public Set<DataverseRole> availableRoles(Long dvId) {
         Set<DataverseRole> roles = dv.getRoles();
         roles.addAll(findBuiltinRoles());
 
-        while (!dv.isEffectivelyPermissionRoot()) {
+        while (dv.getOwner() != null) {
             dv = dv.getOwner();
```

Comment:
So, we're saying here that Dataverses will always inherit roles from their parents. It looks like we never implemented  permission root that would allow the DV owner to refuse to inherit. There are other usages of isEffectivelyPermissionRoot. Should they be removed in favor of getOwner as well?

Reply:
I did run this by @scolapasta to make sure I got it right, and it really appears that the line above was simply a bug. I.e., was a wrong place to apply this "permission root" logic. In other words, yes, the roles themselves (the roles available for assignment) should always be inherited. 
We DO want to continue using the permission root concept elsewhere (even though there doesn't appear to be a way currently to set the boolean to false, other than by changing it in the database). It's just that it is meant for restricting the inheritance of role *assignments*, from the parent dataverse to its descendant dataverses. In other words, it should still be possible to set PermissionRoot to false and have a dataverse/collection automatically inherit the role assignments from the parent. 

> It looks like we never implemented permission root that would allow the DV admin to refuse to inherit.

So, it's the other way around, all our dataverses are "permission roots" as of now and therefore refuse to inherit the assignments. 
It also looks like we DID have it implemented at some point, i.e. there used to be a way to set it to false. There's the `UpdatePermissionRootCommand` and this fragment: `RolePermissionFragment.java` appears to be a remnant of a page where it could be changed (but it's not being used as of now). I didn't remember this, but according to Gustavo we had stopped allowing it because "it was causing too much problems" in some cases. But from what I understand, we want to to keep the code that checks for it around in places like `PermissionServiceBean`. 

Does this make sense? I hope I'm getting it right. This is rather old design history. And also, so much time has passed since I talked to Gustavo about it and made this PR...

Reply:
OK. that makes sense. I recall that we did once allow DVs to set it to false, I just don't remember when that was changed.